geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/ServerConnection.java

```diff
@@ -894,6 +876,50 @@ void doNormalMessage() {
     }
   }
 
+  ThreadState bindSubject(Command command) {
+    if (!securityService.isIntegratedSecurity()) {
+      return null;
+    }
+
+    if (communicationMode.isWAN()) {
+      return null;
+    }
+
+    if (command instanceof PutUserCredentials) {
+      return null;
+    }
+
+    if (isInternalMessage(requestMessage, allowInternalMessagesWithoutCredentials)) {
+      return null;
+    }
+
+    long uniqueId = getUniqueId();
+    String messageType = MessageType.getString(requestMessage.getMessageType());
+    if (uniqueId == 0 || uniqueId == -1) {
+      logger.debug("No unique ID yet. {}, {}", messageType, getName());
```

Comment:
always not sure what's the benefit of that, I was just following what was done before. I hate to clutter up the code with more `if` statements if it's not necessary.

Reply:
I just checked and it seems that wrapping the log statement in that check has no effect on the behaviour. I guess that it was necessary at one point, because I see that pattern all over the place, but it seems that it's fine not to do it here. I learned something new today!